tests/h/services/group_list_test.py

```diff
@@ -380,16 +194,122 @@ def test_groups_are_sorted_alphabetically(
         assert [group.pubid for group in results] == expected_sorted_pubids
 
 
-class TestListGroupsFactory(object):
-    def test_list_groups_factory(self, pyramid_request):
-        svc = list_groups_factory(None, pyramid_request)
+class TestUserGroups(object):
+    def test_it_returns_all_of_a_users_groups(self, svc, user, user_groups):
+        user.groups = user_groups
+
+        u_groups = svc.user_groups(user)
+
+        for group in user.groups:
+            assert group in u_groups
+        assert len(u_groups) == len(user_groups)
+
+    def test_it_sorts_a_users_groups_by_name(self, svc, user, user_groups):
+        user.groups = user_groups
+
+        u_groups = svc.user_groups(user)
+
+        group_names = [group.name for group in u_groups]
+        assert group_names == ["Alpha", "Beta", "Gamma", "Oomph"]
+
+    def test_it_returns_empty_list_if_no_user(self, svc):
+        u_groups = svc.user_groups(user=None)
+
+        assert u_groups == []
+
+
+class TestPrivateGroups(object):
+    def test_it_returns_a_users_private_groups(self, svc, user, user_groups):
+        user.groups = user_groups
+
+        p_groups = svc.private_groups(user)
+
+        for group in p_groups:
+            assert group in user.groups
+            assert group.type == "private"
+
+    def test_it_only_returns_private_groups(self, svc, user, user_groups):
+        user.groups = user_groups
+
+        p_groups = svc.private_groups(user)
+
+        group_names = [group.name for group in p_groups]
+        assert "Oomph" not in group_names
+
+    def test_it_sorts_private_groups_by_name(self, svc, user, user_groups):
```

Comment:
Hmmm. I went into the tests and tried to futz around with this, but it honestly feels like the first test (`test_it_returns_a_users_private_groups`) is testing to make sure the user's private groups _are_ included and the second test (`test_it_only_returns_private_groups`) is testing that non-private groups are not returned. They seem like two valid tests from where I'm sitting. Granted it's been a bit of a weird day and I'm all over the place.

When I tried to combine the tests into one, it didn't feel self-documenting, which is why I'm unsure of the necessity of this change...

Reply:
I think it could be just one test named it_only_returns_a_users_private_groups and test the list of groups returned against the expected list. I don't think there's a need to get that fine grained here.